### ISSUE TITLE
Deep copy of a BufferedInputStream

### ISSUE THREAD

**Issue by pacellig** [kostaskougios/cloning#51]
I tried to use the library in this way

>             Cloner cloner=new Cloner();
>             BufferedInputStream clonedBis = cloner.deepClone(bis);
>             //create a byte array
>             byte[] clonedContents = new byte[1024];
>                 
>             int clonedBytesRead=0;
>             String clonedStrFileContents;
> 
>             while( (clonedBytesRead = clonedBis.read(clonedContents)) != -1){

but the cloned BufferedInputStream doesn't seem to contain any data ( the while loop doesn't execute, since clonedBis.read(clonedContents) is -1.

**Comment by kostaskougios** [comment]
BufferedInputStream uses your original stream to fetch data. If you used a FileInputStream, chances are that the cloned FileInputStream contains a cloned FileDescriptor but with the same FileDescriptor.fd. This must be the low level OS file handler. So the behaviour of it will be unexpected. Maybe you already read the file using your "bis" BufferedInputStream and the cloned returns -1 (since it is the same low level file).

**Comment by pacellig** [comment]
Yes, I did already read the file using 'bis'. There is no way to clone an already read BufferedInputStream?

**Comment by kostaskougios** [comment]
Cloner can't clone the native file handler. 

Also cloning non-java pointers can lead to jvm crashes, so I would recommend avoiding it. I think in this case it is better to just re-open the file if you need to re-read it.

**Comment by Littlenotch-Jacuzzi** [comment]
The stream read operation is fundamentally based on the file handler which gets by the long field <code>handle</code> in class <code>FileDescriptor</code>. It means different streams have different file handlers(even if two stream point to the same file), so the <code>handle</code> field of every stream instance <b>should be</b> different.But clone operation will make two streams have the same file handler, it causes two streams share one <code>read()</code> operation, when one of the stream reaches the end, the other stream's reading operation will return -1.
